src/pages/RegPage.jsx

```diff
@@ -0,0 +1,101 @@
+import React, { useReducer, useState } from "react";
+import DisplayRequest from "../views/DisplayRequest";
+import FlowData from "../views/FlowData";
+import API, { getFormattedURL, getFormattedProxyURL } from "../API";
+import RegistrationProvider from "../util/RegistrationProvider";
+import { Container, Row, Col } from "react-bootstrap";
+import Registration from "../views/Registration";
+
```

Comment:
```suggestion

const initialState = {
    flowID: "",
    csrf: "",
    accountKey: "",
    sessionID: "",
    request: {},
    response: {},
};

```
The state here is quiete complex. State updates should also be done atomically. Every call of a `setState` hook could trigger a rerendering. However, if there's still a running function you could loose the values in the state you just updated. We therefore update the component to use a single state object and update it atomically. See the following suggestions.